# Exercise 2

#### Plan:

Use the CPS-ORG files from the NBER, append the morg files, rename variables, keep certain observation and summarize/describe 

Prepare the CPI data, reshape the data from wide to long and rename and label variables.

Merge the CPI variable into the CPS data, Create a real weekly wage variable in November 2020 dollars.

Create and Plot a data set of statistics.


In [1]:
cd "~/Eco 727 - Research Methods/Analysis"

C:\Users\Rafael\Eco 727 - Research Methods\Analysis


## A. Using the CPS-ORG Files from the NBER

### Using the append command to read the three files into one stacked data set

In [2]:
append using https://data.nber.org/morg/annual/morg79.dta  https://data.nber.org/morg/annual/morg80.dta https://data.nber.org/morg/annual/morg81.dta 

(label prunecat already defined)
(label ptstat already defined)
(label ftpt79 already defined)
(label why35lw already defined)
(label absentlw already defined)
(label reasonlw already defined)
(label uhours35 already defined)
(label doinglw already defined)
(label activlwr already defined)
(label eligible already defined)
(label docc70 already defined)
(label dind already defined)
(label ind70 already defined)
(label esr already defined)
(label classer already defined)
(label class already defined)
(label I25d already defined)
(label I25c already defined)
(label I25b already defined)
(label I25a already defined)
(label paidhre already defined)
(label paidhr already defined)
(label relahh already defined)
(label gradecp already defined)
(label veteran already defined)
(label marital already defined)
(label ethnic already defined)
(label race already defined)
(label sex already defined)
(label smsarank already defined)
(label smsa70 already defined)
(label centcity already defined)
(labe

### Renaming variables for consistency with the IPUMS data

In [3]:
rename intmonth month
label value month MONTH
label define MONTH 1 "january" 2 "february" 3 "march" 4 "april" 5 "may" 6 "june" 7 "july" 8 "august" ///
                   9 "september" 10 "october" 11 "november" 12 "december"

In [4]:
rename minsamp mish
label value mish MISH
label define MISH 4 "four" 8 "eight" // To match the IPUMS format.

In [5]:
tab1 month mish


-> tabulation of month  

  Interview |
      month |      Freq.     Percent        Cum.
------------+-----------------------------------
    january |     90,779        8.42        8.42
   february |     92,027        8.53       16.95
      march |     91,724        8.51       25.46
      april |     92,418        8.57       34.03
        may |     87,494        8.11       42.14
       june |     87,019        8.07       50.21
       july |     90,039        8.35       58.56
     august |     89,386        8.29       66.85
  september |     88,548        8.21       75.06
    october |     88,926        8.25       83.31
   november |     89,585        8.31       91.62
   december |     90,386        8.38      100.00
------------+-----------------------------------
      Total |  1,078,331      100.00

-> tabulation of mish  

   Month in |
sample (4 & |
      8 are |
 departing) |      Freq.     Percent        Cum.
------------+-----------------------------------
       four |    539,

We do a quick tabulation of the newly renamed variables to ensure that it was done correctly. 

### Keeping people: ages 18-65, employed last week & working for a private firm or a government.

In [6]:
numlabel esr, add // To add code number for labels in order to see who is employed.
tab esr




   Employment status recode |      Freq.     Percent        Cum.
----------------------------+-----------------------------------
        1. Employed-At Work |    602,351       55.86       55.86
         2. Employed-Absent |     38,076        3.53       59.39
      3. Unemployed-Looking |     46,858        4.35       63.74
               4. Housework |    211,136       19.58       83.32
                  5. School |     47,934        4.45       87.76
 6. Unable to work/Disabled |     17,676        1.64       89.40
7. Other (Includes Retired) |    114,300       10.60      100.00
----------------------------+-----------------------------------
                      Total |  1,078,331      100.00


In [7]:
numlabel classer, add // To add code number for labels in order to see which class of worker to keep. 
tab classer




                  (e&r) Class of worker |      Freq.     Percent        Cum.
----------------------------------------+-----------------------------------
                             1. Private |    606,926       73.61       73.61
                          2. Government |    133,281       16.16       89.77
                       3. Self-Employed |     69,780        8.46       98.23
                         4. Without Pay |      9,129        1.11       99.34
5. Never worked or never worked full-ti |      5,428        0.66      100.00
----------------------------------------+-----------------------------------
                                  Total |    824,544      100.00


In [8]:
numlabel activlwr, add // To add code number for labels in order to see who was working/employed last week.
tab activlwr




 (r) Major Activity |
          last week |      Freq.     Percent        Cum.
--------------------+-----------------------------------
         1. Working |    560,184       51.95       51.95
      2. With a job |     26,969        2.50       54.45
3. Looking for Work |     18,369        1.70       56.15
   4. Keeping House |    238,672       22.13       78.29
       5. At School |     75,005        6.96       85.24
  6. Unable to Work |     17,688        1.64       86.88
   7. Other/Retired |    141,444       13.12      100.00
--------------------+-----------------------------------
              Total |  1,078,331      100.00


In [9]:
keep if inrange(age, 18,65) // Keeping people between the ages of 18-65.
keep if inlist(esr, 1, 2) // Keeping people whose status is: employed at work and employed-absent
keep if inlist(activlwr, 1, 2) // Keeping people whose major activity was working or with a job last week
keep if inlist(classer, 1, 2) // Keeping people who work for a private firm or goverment. 


(202,180 observations deleted)

(273,844 observations deleted)

(43,657 observations deleted)

(53,353 observations deleted)


### Describe and Summarize

In [10]:
describe


Contains data
  obs:       505,297                          
 vars:            59                          
 size:    44,971,433                          
--------------------------------------------------------------------------------
              storage   display    value
variable name   type    format     label      variable label
--------------------------------------------------------------------------------
mish            byte    %8.0g      MISH       Month in sample (4 & 8 are
                                                departing)
month           byte    %9.0g      MONTH      Interview month
hhid            str12   %12s                  Household ID (12 digits)
state           byte    %8.0g      state      State
smsarank        byte    %35.0g     smsarank   SMSA ranking
hurespli        byte    %8.0g                 Line Number of Respondent
hhnum           byte    %8.0g                 Household number
activlwr        byte    %19.0g     activlwr   (r) Major Activity last

We end up with roughly 500 thousand observations on 59 variables. 

In [11]:
summarize


    Variable |        Obs        Mean    Std. Dev.       Min        Max
-------------+---------------------------------------------------------
        mish |    505,297    5.992397    1.999988          4          8
       month |    505,297     6.48617    3.461904          1         12
        hhid |          0
       state |    505,297    53.29502    26.41593         11         95
    smsarank |    505,297    5.533001      10.745          0         57
-------------+---------------------------------------------------------
    hurespli |    503,468    1.707324     .791189          1          7
       hhnum |    505,297    1.117297    .3735208          1          8
    activlwr |    505,297    1.040742     .197693          1          2
     hourslw |    485,897    40.04876    11.20389          1         99
    reasonlw |     97,075    9.631615    3.926637          1         15
-------------+---------------------------------------------------------
    absentlw |     19,751    2.375627

Based on the summary we can see that that some variables have missing observations, such as earnwke. We also see that interview month and month in sample were properly renamed.

### What's the mean and range of earnwke? tab year if earnwke isn't missing.


In [12]:
summarize earnwke


    Variable |        Obs        Mean    Std. Dev.       Min        Max
-------------+---------------------------------------------------------
     earnwke |    491,404    277.4752    168.3529          0        999


The mean of earnwke is 277.48 with a range of (999 - 0) = 999. However, earnwke most likely has some missing values and is top coded. Where 999 represents weekly earnings of $999 or more, thereby distorting the weekly avereage.   

In [13]:
tab earnwke if earnwke == !.


     (e&dp) |
   Earnings |
   per week |      Freq.     Percent        Cum.
------------+-----------------------------------
          0 |         68      100.00      100.00
------------+-----------------------------------
      Total |         68      100.00


Once we tabulate year if earnwke is not missing, we observe 68 observations which should represent the 68 observations who earned $0 in weekly wages.

In [14]:
save "../Data/nber", replace // We save the data as nber to our Data folder.

file ../Data/nber.dta saved


## B. Preparing the CPI Data

In [15]:
import excel "..\Data\CPI.xlsx", cellrange(A12:M51) firstrow clear

In [16]:
list


     +------------------------------------------------------------------+
  1. | Year |     Jan |     Feb |     Mar |     Apr |     May |     Jun |
     | 1982 |    94.3 |    94.6 |    94.5 |    94.9 |    95.8 |      97 |
     |------------------------------------------------------------------|
     |     Jul  |     Aug  |     Sep  |     Oct  |     Nov  |      Dec  |
     |    97.5  |    97.7  |    97.9  |    98.2  |      98  |     97.6  |
     +------------------------------------------------------------------+

     +------------------------------------------------------------------+
  2. | Year |     Jan |     Feb |     Mar |     Apr |     May |     Jun |
     | 1983 |    97.8 |    97.9 |    97.9 |    98.6 |    99.2 |    99.5 |
     |------------------------------------------------------------------|
     |     Jul  |     Aug  |     Sep  |     Oct  |     Nov  |      Dec  |
     |    99.9  |   100.2  |   100.7  |     101  |   101.2  |    101.3  |
     +------------------------------

     +------------------------------------------------------------------+

     +------------------------------------------------------------------+
 17. | Year |     Jan |     Feb |     Mar |     Apr |     May |     Jun |
     | 1998 |   161.6 |   161.9 |   162.2 |   162.5 |   162.8 |     163 |
     |------------------------------------------------------------------|
     |     Jul  |     Aug  |     Sep  |     Oct  |     Nov  |      Dec  |
     |   163.2  |   163.4  |   163.6  |     164  |     164  |    163.9  |
     +------------------------------------------------------------------+

     +------------------------------------------------------------------+
 18. | Year |     Jan |     Feb |     Mar |     Apr |     May |     Jun |
     | 1999 |   164.3 |   164.5 |     165 |   166.2 |   166.2 |   166.2 |
     |------------------------------------------------------------------|
     |     Jul  |     Aug  |     Sep  |     Oct  |     Nov  |      Dec  |
     |   166.7  |   167.1  |   167.9

     | 233.596  | 233.877  | 234.149  | 233.546  | 233.069  |  233.049  |
     +------------------------------------------------------------------+

     +------------------------------------------------------------------+
 33. | Year |     Jan |     Feb |     Mar |     Apr |     May |     Jun |
     | 2014 | 233.916 | 234.781 | 236.293 | 237.072 |   237.9 | 238.343 |
     |------------------------------------------------------------------|
     |     Jul  |     Aug  |     Sep  |     Oct  |     Nov  |      Dec  |
     |  238.25  | 237.852  | 238.031  | 237.433  | 236.151  |  234.812  |
     +------------------------------------------------------------------+

     +------------------------------------------------------------------+
 34. | Year |     Jan |     Feb |     Mar |     Apr |     May |     Jun |
     | 2015 | 233.707 | 234.722 | 236.119 | 236.599 | 237.805 | 238.638 |
     |------------------------------------------------------------------|
     |     Jul  |     Aug  |     Sep

We list the data to see structure of it. We observe that is wide and we want to reshape it to long.

### Reshaping the data

In [17]:
rename (Jan-Dec) cpi#, addnumber // We replace the name of the month with CPI and we add a number value to it, starting with 1.
reshape long cpi, i(Year) j(month) // We use Year as id and j as the monthly value.
list



(note: j = 1 2 3 4 5 6 7 8 9 10 11 12)

Data                               wide   ->   long
-----------------------------------------------------------------------------
Number of obs.                       39   ->     468
Number of variables                  13   ->       3
j variable (12 values)                    ->   month
xij variables:
                    cpi1 cpi2 ... cpi12   ->   cpi
-----------------------------------------------------------------------------


     +------------------------+
     | Year   month       cpi |
     |------------------------|
  1. | 1982       1      94.3 |
  2. | 1982       2      94.6 |
  3. | 1982       3      94.5 |
  4. | 1982       4      94.9 |
  5. | 1982       5      95.8 |
     |------------------------|
  6. | 1982       6        97 |
  7. | 1982       7      97.5 |
  8. | 1982       8      97.7 |
  9. | 1982       9      97.9 |
 10. | 1982      10      98.2 |
     |------------------------|
 11. | 1982      11        98 |
 12. | 1982

     |------------------------|
201. | 1998       9     163.6 |
202. | 1998      10       164 |
203. | 1998      11       164 |
204. | 1998      12     163.9 |
205. | 1999       1     164.3 |
     |------------------------|
206. | 1999       2     164.5 |
207. | 1999       3       165 |
208. | 1999       4     166.2 |
209. | 1999       5     166.2 |
210. | 1999       6     166.2 |
     |------------------------|
211. | 1999       7     166.7 |
212. | 1999       8     167.1 |
213. | 1999       9     167.9 |
214. | 1999      10     168.2 |
215. | 1999      11     168.3 |
     |------------------------|
216. | 1999      12     168.3 |
217. | 2000       1     168.8 |
218. | 2000       2     169.8 |
219. | 2000       3     171.2 |
220. | 2000       4     171.3 |
     |------------------------|
221. | 2000       5     171.5 |
222. | 2000       6     172.4 |
223. | 2000       7     172.8 |
224. | 2000       8     172.8 |
225. | 2000       9     173.7 |
     |------------------------|
226. | 2

415. | 2016       7   240.628 |
     |------------------------|
416. | 2016       8   240.849 |
417. | 2016       9   241.428 |
418. | 2016      10   241.729 |
419. | 2016      11   241.353 |
420. | 2016      12   241.432 |
     |------------------------|
421. | 2017       1   242.839 |
422. | 2017       2   243.603 |
423. | 2017       3   243.801 |
424. | 2017       4   244.524 |
425. | 2017       5   244.733 |
     |------------------------|
426. | 2017       6   244.955 |
427. | 2017       7   244.786 |
428. | 2017       8   245.519 |
429. | 2017       9   246.819 |
430. | 2017      10   246.663 |
     |------------------------|
431. | 2017      11   246.669 |
432. | 2017      12   246.524 |
433. | 2018       1   247.867 |
434. | 2018       2   248.991 |
435. | 2018       3   249.554 |
     |------------------------|
436. | 2018       4   250.546 |
437. | 2018       5   251.588 |
438. | 2018       6   251.989 |
439. | 2018       7   252.006 |
440. | 2018       8   252.146 |
     |--

After reshaping the data from wide to long, we list it again to confirm the stucture of the data. 

### Renaming the variables

In [18]:
label value cpi CPI
rename Year year 
label value year YEAR
label value month MONTH
label define MONTH 1 "january" 2 "february" 3 "march" 4 "april" 5 "may" 6 "june" 7 "july" 8 "august" ///
                   9 "september" 10 "october" 11 "november" 12 "december"


We add value labels as well as rename variables in order to match those of the IPUMS CPS data set. 

### Sort, describe, summarize and save the data set

In [19]:
sort year month
list year-cpi in 1/5, noobs




  +------------------------+
  | year      month    cpi |
  |------------------------|
  | 1982    january   94.3 |
  | 1982   february   94.6 |
  | 1982      march   94.5 |
  | 1982      april   94.9 |
  | 1982        may   95.8 |
  +------------------------+


We can observe that data is properly sorted by year first followed by months.

In [20]:
describe
summarize



Contains data
  obs:           468                          
 vars:             3                          
 size:         5,148                          
--------------------------------------------------------------------------------
              storage   display    value
variable name   type    format     label      variable label
--------------------------------------------------------------------------------
year            int     %10.0g     YEAR       Year
month           byte    %9.0g      MONTH      
cpi             double  %10.0g     CPI        
--------------------------------------------------------------------------------
Sorted by: year  month
     Note: Dataset has changed since last saved.


    Variable |        Obs        Mean    Std. Dev.       Min        Max
-------------+---------------------------------------------------------
        year |        468        2001    11.26667       1982       2020
       month |        468         6.5    3.455747          1   

We can see that we end up with 468 observations on 3 varaibles. We can observe that the varaibles names were properly renamed.
We can also look at the Min and Max of each varaible to confirm that the data set takes monthly (1-12) CPI values from 1982 -
2020. Last we see that CPI is missing one observation, since we do not cover December 2020.

In [21]:
save "../Data/cpi", replace // We save the data as cpi to our Data folder.

file ../Data/cpi.dta saved


## C. Merge the CPI Variable into the CPS-ORG Data

In [1]:
use "..\Data\cps" // We load the cps data from exercise 1.
global cpindexNov20 = 257.208 // cpi for nov2020
global mwage20=7.25 // federal minimum wage for 2020 


### Sort by year and month. How many observations do we have? 

In [2]:
sort year month
list year-mish in 1/5, noobs
describe




  +---------------------------------------------------------+
  | year   serial     month   hwtfinl         cpsid    mish |
  |---------------------------------------------------------|
  | 1982        5   january         .   1.98110e+13    four |
  | 1982        5   january         .   1.98110e+13    four |
  | 1982        7   january         .   1.98110e+13    four |
  | 1982       10   january         .   1.98010e+13   eight |
  | 1982       10   january         .   1.98010e+13   eight |
  +---------------------------------------------------------+


Contains data from ..\Data\cps.dta
  obs:     6,265,023                          
 vars:            28                          12 Jan 2021 17:29
 size:   607,707,231                          
--------------------------------------------------------------------------------
              storage   display    value
variable name   type    format     label      variable label
-------------------------------------------------------------

We sort by year and month and list the first 5 obeservation to ensure that the sorting was properly done. 
We have 6,265,023 observations on 28 variables. 

### Merging the CPI into the CPS, with the CPS as the master

In [3]:
summarize earnweek


    Variable |        Obs        Mean    Std. Dev.       Min        Max
-------------+---------------------------------------------------------
    earnweek |  6,256,784    642.4712    531.7074        .01    2884.61


In [4]:
// replace year=year-1 // We need to merge this year's price index to the next survey year's wage data. 
merge m:1 year month using "../Data/cpi", keep(match) nogenerate
// replace year=year+1 
describe
summarize


(label MONTH already defined)

    Result                           # of obs.
    -----------------------------------------
    not matched                             0
    matched                         6,265,023  
    -----------------------------------------


Contains data from ..\Data\cps.dta
  obs:     6,265,023                          
 vars:            29                          12 Jan 2021 17:29
 size:   657,827,415                          
--------------------------------------------------------------------------------
              storage   display    value
variable name   type    format     label      variable label
--------------------------------------------------------------------------------
year            int     %8.0g                 survey year
serial          long    %12.0g                household serial number
month           byte    %8.0g      MONTH      month
hwtfinl         double  %12.0g                household weight, basic monthly
cpsid           do

In [5]:
tab earnweek if year == 1982


     weekly |
   earnings |      Freq.     Percent        Cum.
------------+-----------------------------------
          1 |          9        0.01        0.01
          2 |          6        0.00        0.01
          3 |         22        0.01        0.02
          4 |         18        0.01        0.03
          5 |         56        0.03        0.07
          6 |         37        0.02        0.09
          7 |         31        0.02        0.11
          8 |         35        0.02        0.13
          9 |         20        0.01        0.14
         10 |        138        0.08        0.22
         11 |         23        0.01        0.24
         12 |         81        0.05        0.29
         13 |         54        0.03        0.32
         14 |         47        0.03        0.35
         15 |        127        0.08        0.42
         16 |         61        0.04        0.46
         17 |         58        0.03        0.49
         18 |         62        0.04        0.53
     

        166 |        210        0.13       25.27
        167 |        112        0.07       25.33
        168 |        306        0.18       25.52
        169 |         80        0.05       25.56
        170 |      1,164        0.70       26.26
        171 |         92        0.06       26.32
        172 |        211        0.13       26.44
        173 |        138        0.08       26.53
        174 |        199        0.12       26.65
        175 |      1,438        0.86       27.51
        176 |        235        0.14       27.65
        177 |         58        0.03       27.69
        178 |        148        0.09       27.77
        179 |         72        0.04       27.82
        180 |      2,353        1.41       29.23
        181 |         84        0.05       29.28
        182 |        156        0.09       29.37
        183 |         96        0.06       29.43
        184 |        209        0.13       29.56
        185 |        400        0.24       29.80
        186 |       

        334 |        100        0.06       63.83
        335 |        197        0.12       63.95
        336 |        164        0.10       64.04
        337 |         91        0.05       64.10
        338 |        118        0.07       64.17
        339 |         48        0.03       64.20
        340 |        811        0.49       64.69
        341 |         42        0.03       64.71
        342 |        109        0.07       64.78
        343 |         48        0.03       64.81
        344 |        118        0.07       64.88
        345 |        177        0.11       64.98
        346 |        376        0.23       65.21
        347 |         55        0.03       65.24
        348 |         92        0.06       65.30
        349 |         48        0.03       65.32
        350 |      3,478        2.09       67.41
        351 |         57        0.03       67.45
        352 |         83        0.05       67.50
        353 |         39        0.02       67.52
        354 |       

        502 |         28        0.02       87.41
        503 |         27        0.02       87.43
        504 |         35        0.02       87.45
        505 |         34        0.02       87.47
        506 |         32        0.02       87.49
        507 |         22        0.01       87.50
        508 |         63        0.04       87.54
        509 |         22        0.01       87.55
        510 |        131        0.08       87.63
        511 |         30        0.02       87.65
        512 |         57        0.03       87.69
        513 |         29        0.02       87.70
        514 |         18        0.01       87.71
        515 |         69        0.04       87.75
        516 |         34        0.02       87.78
        517 |         22        0.01       87.79
        518 |         25        0.02       87.80
        519 |        102        0.06       87.86
        520 |        668        0.40       88.27
        521 |         22        0.01       88.28
        522 |       

        670 |         73        0.04       94.32
        671 |          5        0.00       94.32
        672 |         26        0.02       94.34
        673 |        151        0.09       94.43
        674 |         10        0.01       94.44
        675 |        142        0.09       94.52
        676 |         10        0.01       94.53
        677 |          8        0.00       94.53
        678 |         10        0.01       94.54
        679 |          9        0.01       94.54
        680 |        124        0.07       94.62
        682 |         14        0.01       94.63
        683 |         11        0.01       94.63
        684 |          5        0.00       94.64
        685 |         18        0.01       94.65
        686 |          8        0.00       94.65
        687 |         15        0.01       94.66
        688 |         15        0.01       94.67
        689 |          8        0.00       94.67
        690 |         51        0.03       94.71
        691 |       

        846 |         14        0.01       97.55
        847 |          4        0.00       97.55
        848 |          1        0.00       97.55
        849 |          2        0.00       97.55
        850 |        248        0.15       97.70
        851 |          2        0.00       97.70
        852 |          3        0.00       97.71
        853 |          4        0.00       97.71
        854 |          2        0.00       97.71
        855 |          4        0.00       97.71
        856 |          1        0.00       97.71
        857 |          3        0.00       97.71
        858 |          1        0.00       97.71
        859 |          2        0.00       97.72
        860 |         18        0.01       97.73
        861 |          3        0.00       97.73
        862 |          3        0.00       97.73
        863 |          3        0.00       97.73
        864 |          8        0.00       97.74
        865 |         87        0.05       97.79
        866 |       

After merging the data, we matched 6,09873 observations. 

In [6]:
save "../Data/cps2", replace

file ../Data/cps2.dta saved


### Creating a real weekly wage variable in November 2020 dollars.

In [7]:
generate real_weekly_wage = ${cpindexNov20}*earnweek/cpi // real weekly wage in Nov2020 dollars
//scalar lcut=$mwage20 * 40/2 // half weekly earnings from 40 hours/week at the 2020 minimum wage
//drop if real_weekly_wage<lcut

summarize real_weekly_wage


(8,239 missing values generated)


    Variable |        Obs        Mean    Std. Dev.       Min        Max
-------------+---------------------------------------------------------
real_weekl~e |  6,256,784     922.358    656.4979    .009882   4591.242


## D. Create and Plot a Data Set of Statistics.

### Collapse the data to a set of annual statistics: the mean, the 10th, 50th, and 90th percentiles of the real weekly wage

In [8]:
collapse (mean) mean=real_weekly_wage (p10) p10=real_weekly_wage (p50) p50=real_weekly_wage (p90) p90=real_weekly_wage, by(year)
describe




Contains data
  obs:            39                          
 vars:             5                          
 size:           702                          
--------------------------------------------------------------------------------
              storage   display    value
variable name   type    format     label      variable label
--------------------------------------------------------------------------------
year            int     %8.0g                 survey year
mean            float   %9.0g                 (mean) real_weekly_wage
p10             float   %9.0g                 (p 10) real_weekly_wage
p50             float   %9.0g                 (p 50) real_weekly_wage
p90             float   %9.0g                 (p 90) real_weekly_wage
--------------------------------------------------------------------------------
Sorted by: year
     Note: Dataset has changed since last saved.


### Plot the four series from 1982 to 2020.

In [ ]:
generate y=year-1
line p10 y || line p50 y || line p90 y || line mean y, ///
legend(on label(1 "10th") label(2 "50th") label(3 "90th") label (4 "Mean") rows(1)) name(Fig0, replace) /// 
title("Real weekly wages 1982 - 2020")
graph export "..\Analysis\Results\real_weekly_wages.pdf", as(pdf) replace

## D. Create and Plot a Data Set of Statistics.

### Collapse the data to a set of annual statistics: the mean, the 10th, 50th, and 90th percentiles of the real weekly wage

In [ ]:
collapse (mean) mean=real_weekly_wage (p10) p10=real_weekly_wage (p50) p50=real_weekly_wage (p90) p90=real_weekly_wage, by(year)
describe

### Plot the four series from 1982 to 2020.

In [ ]:
generate y=year-1
line p10 y || line p50 y || line p90 y || line mean y, ///
legend(on label(1 "10th") label(2 "50th") label(3 "90th") label (4 "Mean") rows(1)) name(Fig0, replace) /// 
title("Real weekly wages 1982 - 2020")
graph export "..\Analysis\Results\real_weekly_wages.pdf", as(pdf) replace

Based on the grapgh above, we can observe that weekly wages have stagnated for the 10th and 50th percentile, whereas for the 90th percentile, they have increased.

### Log difference between the 90th and 10th percentiles of the real weekly wage.

In [ ]:
generate logdiff = log(p90) - log(p10)
line logdiff y, legend( on label (1 "Log difference between 90th & 10th percentile") rows(1)) name(Fig1, replace) ///
title("Log difference of real weekly wages 1982 - 2020")
graph export "..\Analysis\Results\log_real_weekly_wages_.pdf", as(pdf) replace

Based on the grapgh above, we can observe that log difference of the real weekly wage between the 90th and 10th percentile have increased as the years have gone by. 

## Percentiles

In [ ]:
foreach y of numlist 1982 1983/2020 {
      pctile y`y' = real_weekly_wage if year==`y', nq(100)
    }
keep if y1982~=.
keep y1982-y2020
generate percentile=_n
order percentile, first
describe
summarize

In [ ]:
reshape long y, i(percentile) j(year)
rename y real_weekly_wage
sort year percentile
order year percentile real_weekly_wage
format real_weekly_wage %7.2fc

In [ ]:
list

In [ ]:
save "../Data/wage_percentiles", replace